org.goldendict.desktop.yml

```diff
@@ -113,10 +112,13 @@ modules:
       - INCLUDEPATH+="${FLATPAK_DEST}/include/QtWebKit"
       - CONFIG+=release
     build-commands:
-      - sed -i "s,Icon=goldendict,Icon=${FLATPAK_ID}," redist/goldendict.desktop
-      - sed -i "s,<launchable type="desktop-id">goldendict.desktop</launchable>,<launchable type="desktop-id">${FLATPAK_ID}.desktop</launchable>," redist/goldendict.appdata.xml
-      - make -j$(nproc)
-      - make -j$(nproc) install
+      - desktop-file-edit --set-icon=${FLATPAK_ID} redist/goldendict.desktop
+      # Put the version to the metadata
+      - appstream-util modify redist/goldendict.appdata.xml releases "<release version=\"`cat version.txt`\" date=\"`date --iso-8601`\"/>"
+      - sed -i -e "s,<releases>&lt;,<releases><," -e "s,&gt;</releases>,></releases>," redist/goldendict.appdata.xml
```

Comment:
why not do that as part of your patch?

Reply:
Because it varies from version to version?

Reply:
That's why it should be contributed upstream.

Reply:
I realized they write the versions in the build process: https://github.com/goldendict/goldendict/blob/master/.travis.yml
````
  - export VERSION=$(git rev-parse --short HEAD) # linuxdeployqt uses this for naming the file
  - ./linuxdeployqt-continuous-x86_64.AppImage appdir/usr/share/applications/*.desktop -appimage
````
And it just not compatible with hard-code the version in the metadata.xml. So I don't think they will accept this.

Reply:
That's why the appdata should go upstream and have the version written out as part of the build process as well.

Reply:
> That's why the appdata should go upstream and have the version written out as part of the build process as well.

Didn't get it. The version is written in the build process, since they use `linuxdeployqt`. How can I even persuade them to change the build tool?

> have the version written out as part of the build process as well.

The appdata with version is only bundled in the appimage. The only way to "solve" that is to contribute to `linuxdeployqt`??